src/sail_json_backend/json.ml

```diff
@@ -231,20 +232,73 @@ let parse_encdec i mc format =
       end
   | _ -> assert false
 
+let filter_non_operands components =
+  let rec aux acc = function
+    | [] -> List.rev acc
+    | hd :: tl -> if String.trim hd = "spc" then tl else aux (hd :: acc) tl
+  in
+  aux [] components
+
+let extract_operands k regex components =
+  let rec aux acc = function
+    | [] -> List.rev acc
+    | hd :: tl ->
+        if Str.string_match regex hd 0 then (
+          let operand = Str.matched_group 1 hd in
+          let trimmed =
+            try
+              let comma_index = String.index operand ',' in
+              debug_print ("Operand before trimming: " ^ operand);
+              let trimmed_operand = String.sub operand 0 comma_index in
+              debug_print ("Final trimmed operand: " ^ trimmed_operand);
+              trimmed_operand
+            with Not_found -> operand
+          in
+          try
+            let inputl = Hashtbl.find inputs k in
+            if List.mem operand inputl || List.mem trimmed inputl then aux (trimmed :: acc) tl else aux acc tl
+          with Not_found -> aux (operand :: acc) tl
+        )
+        else aux acc tl
+  in
+  aux [] components
+
+let extract_and_map_operands k components =
+  let regex = Str.regexp ".+(\\(.*\\))" in
```

Comment:
Since this regex is only used within `extract_operands`, let's push its definition into that function and remove it from this one entirely.